progress/angular/projects/ng-progress-icons/src/lib/pxb-progress-icon.component.ts

```diff
@@ -0,0 +1,60 @@
+import {Component, Input, OnChanges} from '@angular/core';
+import color from 'color';
+
+@Component({
+    selector: 'pxb-progress-icon',
+    template: `
+        <span class="pxb-progress-icon"
+              [style.flexDirection]="getFlexDirection()"
+              [style.display]="labelPosition === 'center' ? '' : 'inline-flex'">
+            <span *ngIf="showPercentLabel"
+                  class="pxb-progress-icon-label"
+                  [class.inverted]="inverted"
+                  [class.center]="labelPosition === 'center'"
+                  [style.color]="labelColor"
+                  [style.webkitTextStrokeWidth.px]="labelPosition === 'center' ? size / 60 : 0"
+                  [style.fontSize.px]="labelSize || size/4"
+                  [style.display]="labelPosition === 'center' ? '' : 'flex'"
+                  [style.height.px]="labelPosition === 'center' ? size : 'unset'"
+                  [style.width.px]="labelPosition === 'center' ? size : 'unset'"
+                  [style.lineHeight.px]="labelPosition === 'top' || labelPosition === 'bottom' ? 'unset' : size"
+                  [style.position]="labelPosition === 'center' ? 'absolute' : 'unset'"
```

Comment:
could these be moved to a class instead of inline styles?

Reply:
Anything that doesn't use the `size` input can be moved to a class.

Reply:
All inline styles which use the `size` var have been transitioned to classes.